### PR TITLE
Prompt user to install missing and outdated dependencies

### PR DESCRIPTION
**Pull request by lionel-**
Depends on r-lib/rlang#1332.

Uses `rlang::check_installed()` to prompt users about missing dependencies. The custom `action` callback uses pak if installed, or remotes if installed, and otherwise `install.packages()`.

For outdated packages, the prompt is only triggered if the versions do not match. This way we avoid the large overhead of the dependency functions of pak and remotes. Also this avoids having to reinstall from github when pak or remotes does not recognise that a local or dev install is sufficient.

### CODE REVIEW

**Review by hadley — COMMENTED** [review]
Looks good in theory — I'll install the dev version when you merge this and see how it feels in practice.

File: R/package-deps.r [hunk 1]
```diff
@@ -44,6 +44,44 @@ parse_deps <- function(string) {
...
+
+  # Outdated and missing dependencies are installed using pak if
+  # installed. If not, the remotes package is used if installed.
+  # Otherwise `install.packages()` is used as a last resort but this
+  # method does not support Remotes fields.
+  action <- function(pkg, ...) {
+    if (is_installed("pak")) {
+      deps <- pak::pkg_deps(path, upgrade = FALSE)
+      deps <- deps[deps$package %in% pkg, ]
+      pak::pak(deps$ref, ask = FALSE)
```
Comment (gaborcsardi) [inline R/package-deps.r:72]: This does two rounds of dependency lookups, which is not ideal.

**Review by gaborcsardi — COMMENTED** [review]
I guess YMMV, but this seems a bit too aggressive to me. How about just printing out the outdated packages instead, once a session?

File: R/package-deps.r (hunk 1 above)
Comment (lionel-) [reply R/package-deps.r:72]: Do you have a fix to suggest? This is getting the refs of the package names in `pkg` which are potentially Remotes refs, and then installing them. Is there a way to do these two steps in one go?

**Comment by lionel-** [comment]
> I guess YMMV, but this seems a bit too aggressive to me. How about just printing out the outdated packages instead, once a session?

For my usage at least it's very rare that I don't want to install the minimal requirements for Imports and Depends packages. Perhaps we could add an option to control this behaviour if it turns out too annoying? We could first try it out to see how it works in practice.

**Comment by lionel-** [comment]
Merging now so we can test it more easily.

File: R/package-deps.r (hunk 1 above)
Comment (gaborcsardi) [reply R/package-deps.r:72]: Why not just call 
```r
pkg_install("deps::.")
```

Comment (lionel-) [reply R/package-deps.r:72]: To avoid having to reinstall dependencies from github when pak does not recognise that a local or dev install is sufficient.

**Comment by lionel-** [comment]
I got puzzled for a little while because a `local_` function wasn't behaving as expected in a testthat `setup-` file. We could make it work by using the pattern above.

**Pull request merged**